### ISSUE TITLE
[Performance] onnxruntime 3 times lower than pytorch on cpu

### ISSUE THREAD

**Issue by leilakhalili87** [microsoft/onnxruntime#15738]
### Describe the issue

I converted a xlnet model to onnx, and the model is 3 times slower compared to pytorch in centos 7. I tried add_session_config_entry("session.set_denormal_as_zero", "1") based on previous issues. But still same performance.

It is interesting that on mac for the same model I am getting 2 times faster compared to pytorch.

### To reproduce

self.sessopts = onnxruntime.SessionOptions()
self.sessopts.graph_optimization_level = (onnxruntime.GraphOptimizationLevel.ORT_ENABLE_ALL )
self.sessopts.add_session_config_entry("session.set_denormal_as_zero", "1")
self.sessopts.intra_op_num_threads=4
self.session = onnxruntime.InferenceSession(model_filename, sess_options =self.sessopts, providers=[available_provider]
            )

### Urgency

we have a release soon, we were hoping to deploy onnx model

### Platform

Linux

### OS Version

CentOS Linux 7 (Core)

### ONNX Runtime Installation

Built from Source

### ONNX Runtime Version or Commit ID

1.13.1

### ONNX Runtime API

Python

### Architecture

X86

### Execution Provider

Default CPU

### Is this a quantized model?

Yes

**Comment by RyanUnderhill** [comment]
If you `pip install onnxruntime`, is the performance the same as building it yourself?

Just to be sure, by X86 is this 32-bit? Or do you mean X86-64?

**Comment by leilakhalili87** [comment]
it is  X86-64, I didn't build it. I used pip install, would it make a difference with build?
I tried on gpu, it just gave me 5% better inference time

**Comment by RyanUnderhill** [comment]
> it is X86-64, I didn't build it. I used pip install, would it make a difference with build? I tried on gpu, it just gave me 5% better inference time

Ah, it was because you said "ONNX Runtime Installation - Built from Source" in the initial post. That should just be 'pip install onnxruntime' as you didn't build from source.

If you leave these repro steps out, is there any change?

    self.sessopts.graph_optimization_level = (onnxruntime.GraphOptimizationLevel.ORT_ENABLE_ALL )
    self.sessopts.add_session_config_entry("session.set_denormal_as_zero", "1")
    self.sessopts.intra_op_num_threads=4

**Comment by RyanUnderhill** [comment]
Can you specify exactly what model you used and how you converted it?

**Comment by JohnHerry** [comment]
The same problem.  inference on GPU with onnx format model, is far more slower then that with pytorch model.
And, when I want to optimize the speed with sessionoption.execution_mode = onnxruntime.ExecutionMode.ORT_PARALLEL,  the inference will fail, and it seems that some layer weights had been bind to the cuda:0 on the creation of the InferenceSession, So input data binding to other device will not accessable to be computed.

**Comment by leilakhalili87** [comment]
@RyanUnderhill do you think there is an issue in the model? I am a bit surprised because the same model is faster on mac.

**Comment by RyanUnderhill** [comment]
Unfortunately I didn't have any answer. There's a chance it could be similar to this issue (you'd need to set the value to your physical CPU cores): https://github.com/microsoft/onnxruntime/issues/9732

If you can compare the CPU usage of pytorch vs Onnxruntime that might confirm this is the problem.

**Comment by leilakhalili87** [comment]
I tried setting cpu cores with different values, it didn't help.
Thanks for your response though.

**Comment by RyanUnderhill** [comment]
You're welcome!

How did the CPU usage of onnxruntime compare with Pytorch?

**Comment by leilakhalili87** [comment]
I compared the %cpu usage, it is pretty high in onnx-model almost 2 times and doesn't change much with changing the `intra_op_num_threads` value.

<img width="675" alt="Screenshot 2023-05-10 at 6 12 11 PM" src="https://github.com/microsoft/onnxruntime/assets/33406434/b749eaeb-1c52-444d-ba85-bfe9cc17d22a">

**Comment by RyanUnderhill** [comment]
Yeah, CPU usage isn't the issue then. There's something else going on.
